### PR TITLE
(SIMP-4198) Add FAQ for configuring SSH

### PR DESCRIPTION
**Pull request by op-ct**
This patch adds an FAQ clarifying different scenarios for managing SSH
with SIMP.

SIMP-4481 #close #comment Added FAQ for customizing sshd_config
SIMP-4495 #close #comment Added FAQ for customizing ssh_config

### CODE REVIEW

**Comment by op-ct** [comment]
Thanks, @jeannegreulich.  This location was based on the issue, [SIMP-4198: "There should be a FAQ entry regarding SSH configuration settings"](https://simp-project.atlassian.net/browse/SIMP-4198).

File: docs/help/FAQ/Ssh.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to customize the default settings, you must prevent ``ssh::client``
+from declaring them automatically with ``ssh::client::add_default_entry: false``
+and declare ``Host *`` manually with the defined type
+``ssh::client::host_config_entry``:
+
+.. code-block:: puppet
+
+   class{ 'ssh::client': add_default_entry => false }
```
Comment (jeannegreulich) [inline docs/help/FAQ/Ssh.rst]: I would tell them to set ssh::client::add_default_entry: false in hiera.    (I think the directions should be directed to people using the simp scenario .)

**Comment by jeannegreulich** [comment]
Then you should explain the stragedy in the FAQ and put the how tos under the how to section.

File: docs/help/FAQ/Ssh.rst [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+Including `ssh::server` with the default options will manage the server with
+reasonable settings for each host's environment.
+
+
+Configuring ``ssh::server::conf`` from Hiera
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To customize the SSH server, edit the parameters of `ssh::server::conf` using
+Hiera or ENC (Automatic Parameter Lookup).
```
Comment (lnemsick-simp) [inline docs/help/FAQ/Ssh.rst]: ENC = external node classifier, so I don't understand why 'Automatic Parameter Lookup' is in parens aftewards.

File: docs/help/FAQ/Ssh.rst [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+.. NOTE::
+
+    Unlike many SIMP modules, these customizations cannot be made
+    directly with a resource-style class declaration―they *must* be made via
+    APL.
```
Comment (lnemsick-simp) [inline docs/help/FAQ/Ssh.rst]: APL => hieradata.

File: docs/help/FAQ/Ssh.rst [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+
+Managing additional settings with ``sshd_config``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To manage SSH server settings that aren't managed by the SIMP ``ssh`` module,
+use the ``sshd_config`` resource from `augeasproviders_ssh`_.  This is what the
+SIMP ``ssh`` module uses internally to manage the ``/etc/ssh/sshd_config``
+file, and you can use it to set any options ``ssh::server::conf`` doesn't use.
+
```
Comment (lnemsick-simp) [inline docs/help/FAQ/Ssh.rst]: 'doesn't use' -> 'doesn't manage'

File: docs/help/FAQ/Ssh.rst [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+   The examples below feature ``include 'ssh::server'`` and ``include
+   'ssh::client'``, but most SIMP scenarios already include them both via
+   ``ssh``
+
+
+How do I manage settings for the SSH server?
+=======================================================
+
+Including `ssh::server` with the default options will manage the server with
```
Comment (lnemsick-simp) [inline docs/help/FAQ/Ssh.rst]: double back ticks needed.  Currently, ssh::server renders in italics

File: docs/help/FAQ/Ssh.rst (hunk 2 above)
Comment (lnemsick-simp) [reply docs/help/FAQ/Ssh.rst]: double back ticks needed.  Currently, ssh::server::conf renders in italics

**Comment by trevor-vaughan** [comment]
@jeannegreulich I agree that this should be a HowTo instead of a FAQ since it's literally a 'How To X'

**Comment by trevor-vaughan** [comment]
Pushed after rebase

**Pull request merged**